src/js/admin/screens/groupsScreen.js

```diff
@@ -0,0 +1,150 @@
+const tableHeader = `
+  <div class="tr groups-table-header">
+    <div class="td sr-img">Image</div>
+    <div class="td sr-number">Number</div>
+    <div class="td sr-id">Key</div>
+    <div class="td sr-name">Name</div>
+    <div class="td sr-edit">Edit</div>
+  </div>
+`;
+
+App.renderGroupsScreen = () => {
+    const groupKeys = Object.keys(App.groups);
+    const header = $(`
+    <div id="cp-header" class="card-header">
+      <div class="cp-name">Groups</div>
+      <div class="cp-control">${groupKeys.length}&nbsp;<i class="material-icons">storage</i></div>
+    </div>
+    `);
+    App.jControlPanelHeader.replaceWith(header);
+    App.jControlPanelHeader = header;
+    const cpBody = $(`
+    <div>
+        <div class="card-header d-flex justify-content-between align-items-center">
+            <h5>Manage groups</h5>
+            <button class="btn btn-primary"><i class="material-icons">import_export</i>&nbsp;Import/Export</button>
+        </div>
+        <div class="card-body">
+        <p class="card-text">Add new group by entering its key</p>
+        <form id="group-addition">
+          <div class="input-group">
+            <input class="form-control invalid" placeholder="Enter new groups’ key" title="Key 1-20 digits">
+            <button class="btn btn-primary btn-raised"><i class="material-icons">add</i>&nbsp;Add new</button>
+          </div>
+        </form>
+      </div>
+        <div id="groups-table" class="table"></div>
+    </div>
+    `);
+    App.jGroupsContainer = cpBody.find('#groups-table');
+    const form = cpBody.find('#group-addition');
+    const input = form.find('input');
+    
+    input.keyup(function(){
+        const newKey = input.val();
+        if(newKey && !isNaN(newKey) && !groupKeys[parseInt(newKey)]) {
+            input.removeClass("invalid");
+        } else {
+            input.addClass("invalid");
+        }
+    });
+    form.submit((e) => {
+        e.preventDefault();
+        const newKey = input.val();
+        if (newKey && !isNaN(newKey) && !groupKeys[newKey]) {
+          App.showGroupEditForm(newKey);
+        }
+    });
+    
+    App.renderGroupsTable();
+    App.jControlPanelBody.replaceWith(cpBody);
+    App.jControlPanelBody = cpBody;
+}
+
+App.renderGroupsTable = () => {
+    const groupKeys = Object.keys(App.groups);
+    groupKeys.sort((a, b) => App.groups[a]["number"] - App.groups[b]["number"]); // Descending order
+
+    App.jGroupsContainer.empty();
+    App.jGroupsContainer.append(tableHeader);
+    for (let i = 0; i < groupKeys.length; i++) {
```

Comment:
pouzij spis foreach kdyz vis ze nebudes breakovat

Reply:
zkoušel jsem 
for (groupKey in groupKeys) {...}
ale nějak jsem ztratil můj sort, který je něco nadtím. 
Předtím třídím list groupKeys podle order(měl jsem tam number, ale teď je tam order).

Reply:
```js
groupKeys.forEach((key) => {
    const group = App.groups[groupKey];
    //...
});
```